core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -12131,8 +12131,13 @@ public final Map<String, String> getUserClaimValuesWithID(String userID, String[
         if (isUniqueIdEnabled) {
             isUserExists = doCheckExistingUserWithID(userID);
         } else{
-            user = userUniqueIDManger.getUser(userID, this);
-            isUserExists = user != null;
+            if (userStore != null && StringUtils.isNotBlank(userStore.getDomainName())){
+                user = userUniqueIDManger.getUser(userID, this, userStore.getDomainName());
+                isUserExists = user != null;
+            } else{
+                user = userUniqueIDManger.getUser(userID, this);
+                isUserExists = user != null;
```

Comment:
`isUserExists = user != null;`

We can take this line outside the if statement since it's repeated in both if and else blocks.